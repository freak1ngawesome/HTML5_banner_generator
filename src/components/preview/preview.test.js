import React from 'react';
import Preview from './preview';
import renderer from 'react-test-renderer';
import store from '../../store'

describe('Testing <Preview/>', () => {
  const preview = renderer.create(<Preview store={store}/>);

  it('Component has rendered correctly',() => {
    expect(preview.toJSON()).toMatchSnapshot();
  });

  it('Component has only one image',() => {
    expect(preview.root.findAllByProps({id: 'banner_img'}).length).toBe(1);
  });

  it('Component has only one text',() => {
    expect(preview.root.findAllByProps({id: 'banner_text'}).length).toBe(1);
  });

});
