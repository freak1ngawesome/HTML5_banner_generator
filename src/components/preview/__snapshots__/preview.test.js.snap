// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Preview/> Component has rendered correctly 1`] = `
<section
  className="preview"
  id="preview"
>
  <div
    className="banner"
    id="capture"
    style={
      Object {
        "background": "#000000",
        "borderRadius": "10px",
        "height": "376px",
        "overflow": "hidden",
        "position": "absolute",
        "width": "282px",
      }
    }
  >
    <img
      alt=""
      id="banner_img"
      src=""
      style={
        Object {
          "left": "0",
          "position": "absolute",
          "top": "0",
        }
      }
    />
    <div
      id="banner_text"
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "monospace",
          "fontSize": "44px",
          "fontWeight": "700",
          "height": "132px",
          "left": "10px",
          "lineHeight": "44px",
          "overflow": "hidden",
          "position": "absolute",
          "top": "50%",
          "userSelect": "none",
          "width": "100%",
        }
      }
      suppressContentEditableWarning={true}
    >
       Write your awesome text here
    </div>
  </div>
</section>
`;
